File: erastosthenes.js
/**
 * Created by yury on 29/04/15.
 */
primes_sieve = function (limit) {
    var limitn = limit + 1;
    var not_prime = []
    for(var q=0; q < limitn; q++){
        not_prime[q] = true;
    }
    not_prime[0] = false;
    not_prime[1] = false;
    var primes = [];

    for (var i = 2; i < limitn; i++) {
        if (not_prime[i]) {

            for (var j = 2; i * j < limitn; j++) {
                not_prime[i * j] = false;
            }

            primes.push(i)
        }
    }


    return primes
};

t = new Date().getTime();

primes = primes_sieve(9000000);
//console.log(primes.slice(-3));
console.log((new Date().getTime() - t) / 1000);

File: erastosthenes.py
__author__ = 'yury'
import time

def primes_sieve(limit):
    limitn = limit+1
    not_prime = [False] * limitn
    primes = []

    for i in xrange(2, limitn):
        if not_prime[i]:
            continue
        for f in xrange(i*2, limitn, i):
            not_prime[f] = True

        primes.append(i)

    return primes

if __name__ == '__main__':
    # n = raw_input("input limit:\n> ")
    # n = int(n)
    t = time.time()
    primes = primes_sieve(9000000)
    print "primes found: %d" % primes.__len__()
    # print "last 3 is: %s" % primes[-3:]
    print "time is %0.04F" % (time.time() - t)